HACKING.md

```diff
@@ -92,6 +92,129 @@ Package build dependencies for other distributions can be found under the
 
 Go module dependencies are automatically resolved at build time.
 
+### Building the snap with snapcraft
+
+The easiest (though not the most efficient) way to test changes to snapd is to
+build the snapd snap using _snapcraft_ and then install that snapd snap. The
+snapcraft.yaml for the snapd snap is located at ./build-aux/snapcraft.yaml, and
+can be built using snapcraft either in a LXD container or a multipass VM (or
+natively with `--destructive-mode` on a Ubuntu 16.04 host).
+
+Note: Currently, snapcraft's default track of 5.x does not support building the 
+snapd snap, since the snapd snap uses `build-base: core`, which uses Ubuntu 
+16.04 as the base for building and Ubuntu 16.04 is in Extended Security 
+Maintenance (ESM), and as such only is buildable using snapcraft's 4.x channel.
+
+Install snapcraft from the 4.x channel:
+
+```
+sudo snap install snapcraft --channel=4.x
+```
+
+Then run snapcraft:
+
+```
+snapcraft
+```
+
+Now the snapd snap that was just built can be installed with:
+
+```
+snap install --dangerous snapd_*.snap
+```
+
+To go back to using snapd from the store instead of the custom version we 
+installed (since it will not get updates as it was installed dangerously), you
+can either use `snap revert snapd`, or you can refresh directly with 
+`snap refresh snapd --stable --amend`.
+
+Note: It is also sometimes useful to use snapcraft to build the snapd snap for
+other architectures using the remote-build feature, however there is currently a
+bug in snapcraft around using the 4.x channel and using remote-build, where the
+LP job created for the remote-build will attempt to use the 5.x channel instead
+of the 4.x channel. This being tracked at
+https://warthogs.atlassian.net/browse/CRAFT-568. To work-around this until the
```

Comment:
I'm not sure that this is a useful information, if the bug is not public.

Reply:
There isn't a LP bug about this but I suppose I could file one, this was the bug that the starcraft team themselves filed about it, not sure their policy about using LP vs JIRA for bugs

Reply:
well actually in this case it behooved me to just not do anything since the starcraft team already fixed it and the fix is in edge now :-) https://github.com/snapcore/snapcraft/pull/3600

so I will just revise this to explain to use edge and apply one last workaround to make it work (we have to get rid of the symlink nonsense as currently snapcraft from edge doesn't like it)

Reply:
done in https://github.com/snapcore/snapd/pull/11124